xmake/core/sandbox/modules/import/core/sandbox/sandbox.lua

```diff
@@ -47,8 +48,34 @@ function sandbox_core_sandbox.interactive()
     -- bind sandbox environment
 --    setfenv(0, instance._PUBLIC)
 
+    local enable_readline = os.versioninfo().features.readline
+    local replhistory
+    if enable_readline then
+        -- clear history
+        readline.clear_history()
+
+        -- load history
+        replhistory = history.load("replhistory") or {}
+        for _, ln in ipairs(replhistory) do
+            readline.add_history(ln)
+        end
+    end
+
     -- enter interactive mode with this new sandbox
     sandbox.interactive(instance._PUBLIC) 
+
+    if enable_readline then
+        -- save to history
+        local entries = readline.history_list()
+        if #entries > #replhistory then
+            for i = #replhistory+1, #entries do
+                history.save("replhistory", entries[i].line)
+            end
+        end
+
```

Comment:
I think maybe never be reached here, because I often quit `xmake lua` using `<C-C>`.

So I still feel that It should save the global history in the following place:
 
```c
#ifdef XM_CONFIG_API_HAVE_READLINE
    // get line
    tb_char_t const* line = readline(prompt);
    if (line)
    {
        // add line to history
        add_history(line);

       // save current history to file
       // ..
```